crates/bevy_reflect/src/generics.rs

```diff
@@ -0,0 +1,338 @@
+use crate::type_info::impl_type_methods;
+use crate::{Reflect, Type, TypePath};
+use alloc::borrow::Cow;
+use alloc::sync::Arc;
+use core::ops::Deref;
+
+/// The generic parameters of a type.
+///
+/// This is automatically generated via the [`Reflect` derive macro]
+/// and stored on the [`TypeInfo`] returned by [`Typed::type_info`]
+/// for types that have generics.
+///
+/// It supports both type parameters and const parameters
+/// so long as they implement [`TypePath`].
+///
+/// If the type has no generics, this will be empty.
+///
+/// If the type is marked with `#[reflect(type_path = false)]`,
+/// the generics will be empty even if the type has generics.
+///
+/// [`Reflect` derive macro]: bevy_reflect_derive::Reflect
+/// [`TypeInfo`]: crate::type_info::TypeInfo
+/// [`Typed::type_info`]: crate::Typed::type_info
+#[derive(Clone, Default, Debug)]
+pub struct Generics(Box<[GenericInfo]>);
+
+impl Generics {
+    /// Creates an empty set of generics.
+    pub fn new() -> Self {
+        Self(Box::new([]))
+    }
+
+    /// Finds the generic parameter with the given name.
+    ///
+    /// Returns `None` if no such parameter exists.
+    pub fn get_named(&self, name: &str) -> Option<&GenericInfo> {
```

Comment:
Can we add a iter method so the user can see all the generics despite not knowing the name?

Reply:
This is already provided via the
```rs
impl Deref for Generics {
    type Target = [GenericInfo];

    fn deref(&self) -> &Self::Target {
        &self.0
    }
}
```
You can see an example usage in one of the unit tests:
```rs
        let mut iter = generics.iter();

        let t = iter.next().unwrap();
        assert_eq!(t.name(), "T");
        assert!(t.ty().is::<f32>());
        assert!(!t.is_const());
```

Reply:
Missed the Deref impl somehow, good!